homeassistant/components/overkiz/climate_entities/atlantic_electrical_heater_with_adjustable_temperature_setpoint.py

```diff
@@ -0,0 +1,138 @@
+}
+
+HVAC_MODE_TO_OVERKIZ = {v: k for k, v in OVERKIZ_TO_HVAC_MODE.items()}
+
+
+class AtlanticElectricalHeaterWithAdjustableTemperatureSetpoint(
+    OverkizEntity, ClimateEntity
+):
+    """Representation of Atlantic Electrical Heater (With Adjustable Temperature Setpoint)."""
+
+    _attr_hvac_modes = [*HVAC_MODE_TO_OVERKIZ]
+    _attr_preset_modes = [*PRESET_MODE_TO_OVERKIZ]
+    _attr_temperature_unit = TEMP_CELSIUS
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+
+    def __init__(
+        self, device_url: str, coordinator: OverkizDataUpdateCoordinator
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(2)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        if OverkizState.CORE_OPERATING_MODE in self.device.states:
+            state = self.executor.select_state(OverkizState.CORE_OPERATING_MODE)
+            return OVERKIZ_TO_HVAC_MODE[OverkizCommandParam(state)]
+        if OverkizState.CORE_ON_OFF in self.device.states:
+            state = self.executor.select_state(OverkizState.CORE_ON_OFF)
+            return OVERKIZ_TO_HVAC_MODE[OverkizCommandParam(state)]
+
+        return HVACMode.OFF
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target hvac mode."""
+        if OverkizState.CORE_OPERATING_MODE in self.device.states:
+            await self.executor.async_execute_command(
+                OverkizCommand.SET_OPERATING_MODE, HVAC_MODE_TO_OVERKIZ[hvac_mode]
+            )
+        elif hvac_mode == HVACMode.OFF:
+            await self.executor.async_execute_command(OverkizCommand.OFF)
+        else:
+            await self.executor.async_execute_command(
+                OverkizCommand.SET_HEATING_LEVEL, OverkizCommandParam.COMFORT
+            )
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., home, away, temp."""
+        if state := self.device.states[OverkizState.IO_TARGET_HEATING_LEVEL]:
+            return OVERKIZ_TO_PRESET_MODE[OverkizCommandParam(state.value)]
+        return None
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        if preset_mode in [PRESET_AUTO, PRESET_PROG]:
+            command = OverkizCommand.SET_OPERATING_MODE
+        else:
+            command = OverkizCommand.SET_HEATING_LEVEL
+        await self.executor.async_execute_command(
+            command, PRESET_MODE_TO_OVERKIZ[preset_mode]
+        )
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature."""
+        if OverkizState.CORE_TARGET_TEMPERATURE in self.device.states:
+            state = self.executor.select_state(OverkizState.CORE_TARGET_TEMPERATURE)
```

Comment:
maybe self.device.states[OverkizState.CORE_TARGET_TEMPERATURE] ?  
And then value_as_float

Reply:
Done.

Reply:
I've actually made the same comment on the review yesterday, then deleted it because I thought I was wrong ... 🤦‍♂️